platform/viewer/src/routes/routesUtil.js

```diff
@@ -1,105 +1,141 @@
-  )
-);
-const ViewerRouting = asyncComponent(() =>
-  import(/* webpackChunkName: "ViewerRouting" */ './ViewerRouting.js')
-);
-
-const StudyListRouting = asyncComponent(() =>
-  import(
-    /* webpackChunkName: "StudyListRouting" */ '../studylist/StudyListRouting.js'
-  )
-);
-const StandaloneRouting = asyncComponent(() =>
-  import(/* webpackChunkName: "ConnectedStandaloneRouting" */ '../connectedComponents/ConnectedStandaloneRouting.js')
-);
-const ViewerLocalFileData = asyncComponent(() =>
-  import(
-    /* webpackChunkName: "ViewerLocalFileData" */ '../connectedComponents/ViewerLocalFileData.js'
-  )
-);
-
 const reload = () => window.location.reload();
 
-const ROUTES_DEF = {
-  default: {
-    viewer: {
-      path: '/viewer/:studyInstanceUids',
-      component: ViewerRouting,
-    },
-    standaloneViewer: {
-      path: '/viewer',
-      component: StandaloneRouting,
-    },
-    list: {
-      path: ['/studylist', '/'],
-      component: StudyListRouting,
-      condition: appConfig => {
-        return appConfig.showStudyList !== undefined
-          ? appConfig.showStudyList
-          : true;
-      },
-    },
-    local: {
-      path: '/local',
-      component: ViewerLocalFileData,
-    },
-    IHEInvokeImageDisplay: {
-      path: '/IHEInvokeImageDisplay',
-    },
+const defaultRoutesDefinitions = {
+  viewer: {
+    path: '/viewer/:studyInstanceUids',
   },
-  gcloud: {
-    viewer: {
-      path:
-        '/projects/:project/locations/:location/datasets/:dataset/dicomStores/:dicomStore/study/:studyInstanceUids',
-      component: ViewerRouting,
-      condition: appConfig => {
-        return !!appConfig.enableGoogleCloudAdapter;
-      },
-    },
-    list: {
-      path:
-        '/projects/:project/locations/:location/datasets/:dataset/dicomStores/:dicomStore',
-      component: StudyListRouting,
-      condition: appConfig => {
-        const showList =
-          appConfig.showStudyList !== undefined
-            ? appConfig.showStudyList
-            : true;
-
-        return showList && !!appConfig.enableGoogleCloudAdapter;
-      },
-    },
+  standAloneViewer: {
+    path: '/viewer',
+  },
+  studyList: {
+    path: ['/studylist', '/'],
   },
+  local: {
+    path: '/local',
+  },
+  IHEInvokeImageDisplay: {
+    path: '/IHEInvokeImageDisplay',
+  },
+  gcloudViewer: {
+    path:
+      '/projects/:project/locations/:location/datasets/:dataset/dicomStores/:dicomStore/study/:studyInstanceUids',
+  },
+  gcloudList: {
+    path:
+      '/projects/:project/locations/:location/datasets/:dataset/dicomStores/:dicomStore',
+  },
+};
+
+const findRouteModule = (routesModules, templateKey) => {
+  for (let routesModuleKey in routesModules) {
+    const _module = routesModules[routesModuleKey].module;
+    if (_module && templateKey in _module) {
+      return _module[templateKey];
+    }
+  }
+};
+
+const pathUniquenessValidation = (routesDefinitions, routesValidationMode) => {
```

Comment:
If these methods are reusable and something we would expect other viewer implementations to accommodate, we may want to think of a good way to package them into `@ohif/core` for consumption, and unit test them.

Reply:
ok done.. added some unit tests also